### PR TITLE
Add Landed Conditions; shut off after a minute + stop recording

### PR DESCRIPTION
**Pull request by AidanJSmith**

### CODE REVIEW

**Comment by AidanJSmith** [comment]
I figure we can just use IMU data to check for a lack of momentum, and that should suffice, no?

File: master_startup.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
 import BMP280 as barometer
 import RPi.GPIO as io
 from imu_utils import IMU
+from subprocess import call
```
Comment (neilbalch) [inline master_startup.py]: I'd just say `subprocess.call()`. It's like `using namespace std;` in C++...

File: master_startup.py [hunk 2]
```diff
@@ -176,7 +178,7 @@ def get_events_list(self):
 
     print("Started recording to file: " + str(video_file_suffix) + ".h264 ...")
     cam.start_recording(
-        video_folder + str(video_file_suffix) + ".mp4", format="h264")
+        video_folder + str(video_file_suffix) + ".mp4", format="h264") # The only question is, does this automatically apply asynchronously? Might want to look into that.
```
Comment (neilbalch) [inline master_startup.py]: Yes, it does. I've tested that.

File: master_startup.py [hunk 3]
```diff
@@ -205,20 +207,25 @@ def get_events_list(self):
...
         elif current_flight_state == FlightState.LANDED:
             # Set zero power.
             esc_pwm.ChangeDutyCycle(ESC_MIN_DUTY)
 
             if datetime.datetime.now() > landed_time + datetime.timedelta(minutes=1):
                 # Begin shutdown procedure
+                cam.stop_recording() 
                 imu_data.add_event("FTS automatic trigger: 10s from launch")
                 log_file.writerow(imu_data.formatted_for_log())
+                time.sleep(10) # Not certain how the module works -- frankly, giving it some time to encode seems sapient
```
Comment (neilbalch) [inline master_startup.py]: Pretty quick, so the 10s delay should be extraneous, but a good idea. We'll leave it in for now. Put a TODO in to see if this is necessary.

File: master_startup.py (hunk 2 above)
Comment (AidanJSmith) [reply master_startup.py]: Okay, nice.

File: master_startup.py (hunk 1 above)
Comment (AidanJSmith) [reply master_startup.py]: Ah, sure-- I was only familiar with this, there's an easy fix.

File: master_startup.py (hunk 3 above)
Comment (AidanJSmith) [reply master_startup.py]: Sounds good. Everything else look fine?

File: master_startup.py [hunk 4]
```diff
@@ -205,20 +207,25 @@ def get_events_list(self):
             # Set full power.
             esc_pwm.ChangeDutyCycle(ESC_MAX_DUTY)
 
-            # TODO: What condition trips this state into LANDED?
-            # landed_time = datetime.datetime.now()
+            # If the acceleration magnitude is next to nothing following the freefall, set state to LANDED
+            if imu_data.get_acc_magnitude() < .1 * 9.81 && imu_data.get_acc_magnitude() > -.1 * 9.81 # Accounting random noise
```
Comment (neilbalch) [inline master_startup.py]: `&&` in C++ is `and` in Python. This could be written cleaner with `abs()`...

File: master_startup.py (hunk 3 above)
Comment (AidanJSmith) [reply master_startup.py]: (I'll make edits in accordance with these comments before merging, if you're fine with that)

File: master_startup.py (hunk 4 above)
Comment (AidanJSmith) [reply master_startup.py]: Oh man, too many languages, recently... good catch.

**Comment by AidanJSmith** [comment]
Alright, fixes applied.

File: master_startup.py [hunk 5]
```diff
@@ -176,7 +177,7 @@ def get_events_list(self):
 
     print("Started recording to file: " + str(video_file_suffix) + ".h264 ...")
     cam.start_recording(
-        video_folder + str(video_file_suffix) + ".mp4", format="h264")
+        video_folder + str(video_file_suffix) + ".mp4", format="h264")
```
Comment (neilbalch) [inline master_startup.py]: Extra whitespace kills lives every day, you know?

File: master_startup.py [hunk 6]
```diff
@@ -205,20 +206,26 @@ def get_events_list(self):
             # Set full power.
             esc_pwm.ChangeDutyCycle(ESC_MAX_DUTY)
 
-            # TODO: What condition trips this state into LANDED?
-            # landed_time = datetime.datetime.now()
+            # If the acceleration magnitude is next to nothing following the freefall, set state to LANDED
+            if imu_data.get_acc_magnitude() < abs(.1 * 9.81)# Accounting random noise
```
Comment (neilbalch) [inline master_startup.py]: `abs()` on the other side, man. 😊 Think about what we're trying to do here.

**Pull request merged**